packages/plugins/sns/src/aws-sns/index.ts

```diff
@@ -0,0 +1,74 @@
+import { SNS, Credentials } from 'aws-sdk';
+
+class AwsSNS {
+  awsSNS: SNS
+
+  init() {
+    const SNSConfigOptions:SNS.ClientConfiguration = {
+      credentials: new Credentials(process.env['RLN_SES_AWS_ACCESS_KEY_ID'], process.env['RLN_SES_AWS_SECRET_ACCESS_KEY']),
+      region: process.env['RLN_SES_AWS_REGION'],
+    };
+    this.awsSNS = new SNS(SNSConfigOptions);
+  }
+
+  /*
+  async sendEmail(
+    sender: string,
+    receivers: string[],
+    subject: string,
+    content: string,
+    awsSendEmailParams?: SES.Types.SendEmailRequest,
+  ): Promise<SES.Types.SendEmailResponse> {
+    const params: SES.Types.SendEmailRequest = {
+      Source: sender,
+      Destination: {
+        ToAddresses: receivers,
+      },
+      Message: {
+        Subject: {
+          Charset: 'UTF-8',
+          Data: subject,
+        },
+        Body: {
+          Html: {
+            Charset: 'UTF-8',
+            Data: content,
+          },
+        },
+      },
+      ...awsSendEmailParams,
+    };
+    return await this.awsSES.sendEmail(params).promise();
+  }
+   */
+
+  async createTopic(topicName: string, displayName: string = topicName) {
+    const params: SNS.Types.CreateTopicInput = {
+      Name: topicName,
+      Attributes:{
+        DisplayName: displayName,
+      },
+    };
+    return await this.awsSNS.createTopic(params).promise();
+  }
+
+  async publish(message: string, ARN: string, messageTitle: string){
+    const params: SNS.Types.PublishInput = {
+      MessageStructure: 'JSON',
+      Message: JSON.stringify({
+        default: message,
+        title: messageTitle,
+      }),
+      TopicArn: ARN,
+    };
```

Comment:
Here we should change for sure.
This is the template I'm using right now:
```{
  "default": "Nuova Promozione",
  "APNS": "{\"aps\":{\"alert\": {\"title\": \"MerchantName\", \"body\": \"Nuova Promozione!\"}, \"sound\": \"bingbong.aiff\"}, \"eventId\": \"6044aa87ac6850ac512096af\" }",
  "GCM": "{ \"data\": { \"title\": \"MerchantName\", \"body\": \"Nuova Promozione!\", \"vibrate\": true, \"eventId\": \"6044aa87ac6850ac512096af\" } }"
}
```
So I think we should specify we are using a custom template to AWS

Reply:
For the custom template we are using MessageStructure: 'json' and we are defining the message with JSON.stringify, we added APNS and GCM objects but we don't know if that's the correct approach.